File: back-end/norlan-client-portal-master/src/main/java/it/norlan/clientportal/model/MaterialeDidattico.java
package it.norlan.clientportal.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

/**
 * Entità JPA per la gestione dell'archivio didattico.
 * Associa i percorsi fisici dei file (es. slide, dispense in PDF) a uno specifico
 * Corso di Formazione, rendendoli disponibili per il download agli iscritti.
 */

@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Entity
@Table(name = "materiale_didattico")
public class MaterialeDidattico {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idMateriale;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_corso", nullable = false)
    private CorsoFormazione corso;

    @Column(name = "titolo_documento", nullable = false)
    private String titoloDocumento;

    @Column(name = "percorso_file", nullable = false)
    private String percorsoFile;

    @Column(name = "data_caricamento")
    private LocalDateTime dataCaricamento = LocalDateTime.now();
}

File: front-end/Norlan-Svelte-TypeScript-master/src/lib/models/MaterialeDidattico.ts
/*
Entità per la gestione delle risorse e dei documenti associati ai corsi di formazione.
Mantiene i riferimenti ai file caricati, collegandoli al corso specifico e alla data di upload.
*/
export interface MaterialeDidatticoData {
	idMateriale: number;
	idCorso: number;
	titoloCorso: string;
	titoloDocumento: string;
	percorsoFile: string;
	dataCaricamento: string;
}

export class MaterialeDidattico {
	idMateriale: number;
	idCorso: number;
	titoloCorso: string;
	titoloDocumento: string;
	percorsoFile: string;
	dataCaricamento: string;

	constructor(data: MaterialeDidatticoData) {
		this.idMateriale = data.idMateriale;
		this.idCorso = data.idCorso;
		this.titoloCorso = data.titoloCorso;
		this.titoloDocumento = data.titoloDocumento;
		this.percorsoFile = data.percorsoFile;
		this.dataCaricamento = data.dataCaricamento;
	}
}
